scala/scala-impl/test/org/jetbrains/plugins/scala/lang/refactoring/anonymousToInner/ScalaAnonymousToInnerTest.scala

```diff
@@ -0,0 +1,152 @@
+    val before =
+      s"""def parse(input: Iterator[Byte], someNum: Int): Option[Iterator[Int]] =
+         |  if (input.hasNext) {
+         |    Some(new Ite${Caret}rator[Int] {
+         |      def hasNext = input.hasNext
+         |
+         |      def next = someNum
+         |    })
+         |  } else
+         |    None
+         |""".stripMargin
+    val after =
+      """def parse(input: Iterator[Byte], someNum: Int): Option[Iterator[Int]] =
+        |  if (input.hasNext) {
+        |    Some(new IntIterator(input, someNum))
+        |  } else
+        |    None
+        |
+        |class IntIterator(input: Iterator[Byte], someNum: Int) extends Iterator[Int] {
+        |  def hasNext = input.hasNext
+        |
+        |  def next = someNum
+        |}""".stripMargin
+
+    val className = "IntIterator"
+
+    doTest(before, after, className)
+  }
+
+  def testNestedAnonymousClasses(): Unit = {
+    val before =
+      s"""def parse(input: Iterator[Byte], sliceSize: Int): Option[Iterator[Byte]] =
+         |  if (input.hasNext) {
+         |    Some(new Iterator[Byte] {
+         |      val secondLevelIterator = new Iter${Caret}ator[Int] {
+         |
+         |        override def hasNext: Boolean = true
+         |
+         |        override def next(): Int = sliceSize
+         |      }
+         |
+         |      def hasNext = input.hasNext
+         |
+         |      def next = sliceSize.toByte
+         |    })
+         |  } else {
+         |    None
+         |  }""".stripMargin
+    val after =
+      """def parse(input: Iterator[Byte], sliceSize: Int): Option[Iterator[Byte]] =
+        |  if (input.hasNext) {
+        |    Some(new Iterator[Byte] {
+        |      val secondLevelIterator = new IntIterator(sliceSize)
+        |
+        |      def hasNext = input.hasNext
+        |
+        |      def next = sliceSize.toByte
+        |    })
+        |  } else {
+        |    None
+        |  }
+        |
+        |class IntIterator(sliceSize: Int) extends Iterator[Int] {
+        |
+        |  override def hasNext: Boolean = true
+        |
+        |  override def next(): Int = sliceSize
+        |}""".stripMargin
+
+    val className = "IntIterator"
+
+    doTest(before, after, className)
+  }
+
+  private def doTest(initialText: String, expectedText: String, className: String): Unit = {
+    val handler = new ScalaAnonymousToInnerHandler
+
+    scalaFixture.configureFromFileText(initialText)
+
+    val editor = CommonDataKeys.EDITOR.getData(DataManager.getInstance().getDataContextFromFocusAsync.blockingGet(5, TimeUnit.SECONDS))
+    val caretModel = editor.getCaretModel
+    assert(caretModel.getCaretCount == 1, "Expected exactly one caret.")
+    assert(caretModel.getOffset > 0, s"Not specified caret marker in test case. Use <caret> in scala file for this.")
+
+    val scalaFile = getFile.asInstanceOf[ScalaFile]
+
+    val myElement = scalaFile.findReferenceAt(caretModel.getOffset).getElement
```

Comment:
`my` prefix is used for java fields.
Just `element` is enough.

Reply:
Also this can be shorter:
`val element = scalaFile.findElementAt(caretModel.getOffset)`